---
title: "<NuxtErrorBoundary>"
description: Компонент <NuxtErrorBoundary> обрабатывает ошибки на стороне клиента, возникающие в его слоте по умолчанию.
links:
  - label: Исходники
    icon: i-simple-icons-github
    to: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/components/nuxt-error-boundary.vue
    size: xs
---

::tip
`<NuxtErrorBoundary>` использует под капотом хук Vue [`onErrorCaptured`](https://ru.vuejs.org/api/composition-api-lifecycle.html#onerrorcaptured).
::

## События

- `@error`: Событие генерируется, когда слот по умолчанию компонента выдает ошибку.

  ```vue
  <template>
    <NuxtErrorBoundary @error="logSomeError">
      <!-- ... -->
    </NuxtErrorBoundary>
  </template>
  ```

## Слоты

- `#error`: Укажите резервный контент, который будет отображаться в случае ошибки.

  ```vue
    <template>
      <NuxtErrorBoundary>
        <!-- ... -->
        <template #error="{ error, clearError }">
          <p>An error occurred: {{ error }}</p>

          <button @click="clearError">Clear error</button>
        </template>
      </NuxtErrorBoundary>
    </template>
  ```

:read-more{to="/docs/getting-started/error-handling"}

## Examples

### Accessing `error` and `clearError` in script

You can access `error` and `clearError` properties within the component's script as below:

```vue
<template>
  <NuxtErrorBoundary ref="errorBoundary">
    <!-- ... -->
  </NuxtErrorBoundary>
</template>

<script setup lang="ts">
const errorBoundary = useTemplateRef('errorBoundary')

// errorBoundary.value?.error
// errorBoundary.value?.clearError()
</script>
```
